Sources/bitrise-step-swift-clone/Step.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+import GitKit
+import TSCBasic
+
+struct SwiftCloneStep {
+    let config: Config
+
+    var gitCloneArgs: [String] {
+        var args = [config.repositoryURL, "--verbose"]
+        if let checkoutBranch = config.checkoutBranch {
+            args.append(contentsOf: ["--branch", checkoutBranch])
+        }
+        if let cloneDepth = config.cloneDepth {
+            args.append(contentsOf: ["--depth", cloneDepth])
+        }
+        return args
+    }
+
+    func createClonePath() throws -> AbsolutePath {
+        let currentPath = AbsolutePath(FileManager.default.currentDirectoryPath)
+        let clonePath = currentPath.appending(component: config.cloneDestination)
+        try FileManager.default.createDirectory(atPath: clonePath.pathString, withIntermediateDirectories: true)
+        return clonePath
+    }
+
+    func cloneRepo(into destination: AbsolutePath) throws {
```

Comment:
should this live on the step? or on a FileSystem or Git struct?

Reply:
Open to discussion here, but I think that would make sense if the Step were more complex — i.e. if it were doing more Git operations than just cloning. As it is though, I think it's fine where it is. What do you think?